test/test-platform/CMakeLists.txt

```diff
@@ -26,18 +26,20 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-add_executable(test-platform
-    main.cpp
-    TestPlatform.cpp)
-
-# FIXME: Supress the need for the -Wno-unused-parameter
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unused-parameter")
-
-include_directories(
-    "${PROJECT_SOURCE_DIR}/parameter/include"
-    "${PROJECT_SOURCE_DIR}/remote-processor"
-    "${PROJECT_SOURCE_DIR}/utility")
-
-target_link_libraries(test-platform parameter remote-processor)
-
-install(TARGETS test-platform RUNTIME DESTINATION bin)
+if (BUILD_TESTING)
```

Comment:
don't condition the build of test-platform: it can be used for something else than testing the parameter-framework